tests/fsharpqa/Source/Conformance/Expressions/DataExpressions/ObjectExpressions/E_InvalidSelfReferentialStructConstructor.fs

```diff
@@ -2,8 +2,8 @@
 // FSB 6350, Compiler crash on invalid code: Crash by invalid type definition
 
 //<Expects status="error" span="(19,39)" id="FS0658">Structs may only bind a 'this' parameter at member declarations$</Expects>
-//<Expects status="error" span="(20,14)" id="FS0039">The field, constructor or member 'dt' is not defined</Expects>
 //<Expects status="error" span="(20,9)" id="FS0696">This is not a valid object construction expression\. Explicit object constructors must either call an alternate constructor or initialize all fields of the object and specify a call to a super class constructor\.$</Expects>
+//<Expects status="error" span="(20,14)" id="FS0039">The type 'byref<T,Kind>' does not define the field, constructor or member 'dt'</Expects>
```

Comment:
We probably do not want to show `byref<T,Kind>` as it is a little confusing. Before printing the error, you want to call `destByrefTy` on `ty` if it's a byref type (`isByrefTy` will check if it's a `byref`, `inref`, `outref`).

Based on this specific test, this should be: `The type 'X' does not define the field, constructor or member 'dt'`.